### PR TITLE
drivers: hwinfo: add wakeup flag cause

### PR DESCRIPTION
**Pull request by Desvauxm-st**
add retrieve wakeup flag of device and Clear wakeup flag

### CODE REVIEW

**Review by erwango — CHANGES_REQUESTED** [review]
Please squash the 3 first commits.
Otherwsie LGTM

**Review by erwango — DISMISSED** [review]

**Review by gautierg-st — DISMISSED** [review]

**Review by FRASTM — DISMISSED** [review]

File: drivers/hwinfo/hwinfo_stm32.c [hunk 1]
```diff
@@ -110,7 +116,7 @@ int z_impl_hwinfo_get_reset_cause(uint32_t *cause)
 int z_impl_hwinfo_clear_reset_cause(void)
 {
 	LL_RCC_ClearResetFlags();
-
+	LL_PWR_ClearFlag_SB();
```
[the remark this answers is not in the document]
Comment (Desvauxm-st) [reply drivers/hwinfo/hwinfo_stm32.c:120]: OK

**Review by erwango — DISMISSED** [review]

**Review by gautierg-st — DISMISSED** [review]

**Review by FRASTM — DISMISSED** [review]

**Comment by alexanderwachter** [comment]
@Desvauxm-st merge commits are not allowed in Zephyr. You need to rebase your commits on main instead.

**Review by FRASTM — APPROVED** [review]

**Review by gautierg-st — APPROVED** [review]

**Comment by KloudJack** [comment]
Since this PR was merged, unit tests for STM32 hwinfo now fail.

See the [workflow result](https://github.com/zephyrproject-rtos/zephyr/pull/56330/checks?check_run_id=12397402575) for a PR I've rebased recently.

**Pull request merged**